### PR TITLE
migration and validation on users

### PR DESCRIPTION
**Pull request by carlnoval**
Was meant to only be for migration but went ahead and implemented everything: enum for roles, seeding, user signup form.

Now we can do methods like this for our user model
```ruby
[1] pry(main)> q = User.last
  User Load (0.8ms)  SELECT "users".* FROM "users" ORDER BY "users"."id" DESC LIMIT $1  [["LIMIT", 1]]
=> #<User id: 27, email: "rshante@email.com", created_at: "2022-01-25 16:03:10", updated_at: "2022-01-25 16:03:10", first_name: "Shante", last_name: "Johnson", prefecture: nil, role: "renter">
[2] pry(main)> q.renter?
=> true
[3] pry(main)> q.provider?
=> false
[4] pry(main)> User.renter
  User Load (1.0ms)  SELECT "users".* FROM "users" WHERE "users"."role" = $1  [["role", "renter"]]
=> [#<User id: 21, email: "rhiro@email.com", created_at: "2022-01-25 16:03:08", updated_at: "2022-01-25 16:03:08", first_name: "Hiro", last_name: "Takemura", prefecture: nil, role: "renter">,
 #<User id: 23, email: "rcedrine@email.com", created_at: "2022-01-25 16:03:09", updated_at: "2022-01-25 16:03:09", first_name: "Cedrine", last_name: "Monnet", prefecture: nil, role: "renter">,
 #<User id: 25, email: "rcarl@email.com", created_at: "2022-01-25 16:03:09", updated_at: "2022-01-25 16:03:09", first_name: "Carl", last_name: "Noval", prefecture: nil, role: "renter">,
 #<User id: 27, email: "rshante@email.com", created_at: "2022-01-25 16:03:10", updated_at: "2022-01-25 16:03:10", first_name: "Shante", last_name: "Johnson", prefecture: nil, role: "renter">]
[5] pry(main)> User.provider
  User Load (1.0ms)  SELECT "users".* FROM "users" WHERE "users"."role" = $1  [["role", "provider"]]
=> [#<User id: 20, email: "hiro@email.com", created_at: "2022-01-25 16:03:08", updated_at: "2022-01-25 16:03:08", first_name: "Hiro", last_name: "Takemura", prefecture: nil, role: "provider">,
 #<User id: 22, email: "cedrine@email.com", created_at: "2022-01-25 16:03:08", updated_at: "2022-01-25 16:03:08", first_name: "Cedrine", last_name: "Monnet", prefecture: nil, role: "provider">,
 #<User id: 24, email: "carl@email.com", created_at: "2022-01-25 16:03:09", updated_at: "2022-01-25 16:03:09", first_name: "Carl", last_name: "Noval", prefecture: nil, role: "provider">,
 #<User id: 26, email: "shante@email.com", created_at: "2022-01-25 16:03:09", updated_at: "2022-01-25 16:03:09", first_name: "Shante", last_name: "Johnson", prefecture: nil, role: "provider">]
```

### CODE REVIEW

**Comment by hiroya210** [comment]
Hey man, there are 3 conflicts here, 2 of them are good, but in `app/views/devise/registrations/new.html.erb` can we use the master's branch `:collection`? instead of just **Tokyo and Ibaraki**

**Comment by carlnoval** [comment]
Yeah 3 conflicts, it is weird that I already tried to resolve the `:collection` one earlier but I guess gh does not like it that I only resolve one. Maybe I have to resolve all at once. I'm actually not sure on what to do about the migration one so I'll ask some assistance on this from teachers tomorrow.

Removing **Tokyo and Ibaraki** was actually what I did earlier to resolve.

**Comment by carlnoval** [comment]
Resolved my conflicts.
Internal ones? I don't think that could ever be resolved

File: db/seeds.rb [hunk 1]
```diff
@@ -45,7 +44,7 @@
               details: Faker::Lorem.sentence(word_count: 50, supplemental: true, random_words_to_add: 50),
               prefecture: ["Tokyo", "Kanagawa", "Chiba", "Saitama", "Ibaraki", "Tochigi", "Yamanashi"].sample,
               price: ((rand(5..20)) * 1000),
-              user_id: [user1, user2, user3, user4].sample.id
+              user_id: [user1, user3, user5, user6].sample.id
```
Comment (spioupiou) [inline db/seeds.rb:47]: I might be completely wrong but here maybe you need user1, user3, user5 and user7 - not user6. user7 is Shante as a provider whereas user6 is Carl as a renter, and for Kondo creation it should probably sample among the providers only?

Comment (carlnoval) [reply db/seeds.rb:47]: Thanks for catching that Cédrine. I also noticed that the comments I made were wrong. Will make the changes.

**Comment by carlnoval** [comment]
Having troubles with git, I'll try making a new PR.

**Pull request closed without merging**